solidity/contracts/tbtc/Vault.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.4;
+
+import "../bank/Bank.sol";
+import "../token/TBTC.sol";
+
+/// @title TBTC application vault
+/// @notice TBTC is a fully Bitcoin-backed ERC-20 token pegged to the price of
+///         Bitcoin. It facilitates Bitcoin holders to act on the Ethereum
+///         blockchain and access the decentralized finance (DeFi) ecosystem.
+///         Vault mints and redeems TBTC based on Bitcoin balances in the
+///         Bank.
+/// @dev Vault is the owner of TBTC token contract and is the only contract
+///      minting the token.
+contract Vault {
```

Comment:
What do you think about naming it `TBTCVault` and placing that contract in `contracts/vault`. This can make sense since we'll introduce `IVault`. Also, in case we ever want to add a new token & vault pair, that structure seems to be more clear because we would have the token placed in `contracts/token` and the vault in `contracts/vault`. A bare `Vault` looks very generic.

Reply:
Not a bad idea and I do agree bare `Vault` looks too generic. Done in https://github.com/keep-network/tbtc-v2/pull/87/commits/acdf8fe36f98f058a298c45d8761546bf8a57df6 and PR description is now updated as well.